### ISSUE TITLE
Freeze on mp3 with png cover art

### ISSUE THREAD

**Issue by j-b-m** [mltframework/mlt#895]
Mp3 files with a png as cover art cause a playback loop and freeze in MLT.
Reproducible on Linux with MLT v 7.14.0 and latest git master. Tested against FFmpeg 4.4.2 and 5.1.3

Attached is a sample file reproducing the freeze (renamed to .mp4 to allow attaching it)
just playing it freezes at start:

`melt cover-art.mp3`

File was created from a simple ffmpeg command line:

`ffmpeg -i audio.wav -i image.png -map 0:a -map 1:0 file:cover-art.mp3`

But you can easily find podcasts on the net that have the same issue, for example:
[https://www.rts.ch/la-1ere/programmes/tout-un-monde/podcast/](https://www.rts.ch/la-1ere/programmes/tout-un-monde/podcast/)

I could fix the freeze with this patch, but it is probably not the best fix:

```
diff --git a/src/modules/avformat/producer_avformat.c b/src/modules/avformat/producer_avformat.c
index 944ab737..af64f6c4 100644
--- a/src/modules/avformat/producer_avformat.c
+++ b/src/modules/avformat/producer_avformat.c
@@ -2189,6 +2189,12 @@ static int producer_get_image(mlt_frame frame,
                                         self->video_send_result);
                     } else {
                         int error = avcodec_receive_frame(self->video_codec, self->video_frame);
+                        if (error == AVERROR(EAGAIN)) {
+                            while (error == AVERROR(EAGAIN)) {
+                                self->video_send_result = avcodec_send_packet(self->video_codec, &self->pkt);
+                                error = avcodec_receive_frame(self->video_codec, self->video_frame);
+                            }
+                        }
                         if (error < 0) {
                             if (error != AVERROR(EAGAIN) && ++decode_errors > 10) {
                                 mlt_log_warning(MLT_PRODUCER_SERVICE(producer),
```

https://user-images.githubusercontent.com/108252/228807850-8c3266e6-56bb-4674-9ff3-991ec5d89c9e.mp4

**Comment by ddennedy** [comment]
I have not reproduced it with FFmpeg v5.2 or 6 with my own mp3 with art and with the attached sample. Maybe it is a problem specific to those older versions of FFmpeg.
We cannot add an infinite loop on EAGAIN. Please remove the unnecessary `if` and try to add a counter-based retry with a limit that resets on success.

**Comment by j-b-m** [comment]
This is strange, because I can reproduce also with FFmpeg 6.0 on my laptop.
I pushed a better patch here:
https://github.com/j-b-m/mlt/commit/417bf4769da19178ddfae48108daaf82cbeb25e6

I used the same counter as for other decoding errors, and the value of 16 retries is based on my tests, lower values don't fix the freeze.

I can also reproduce the freeze with my sample file on Shotcut's v22.03 ALPHA Applimage (which uses FFmpeg 5.1.2).
I can create an MR if you want but I am still puzzled as why you cannot reproduce...

**Comment by ddennedy** [comment]
Sorry, I meant FFmpeg 5.1.2 instead of 5.2 . I did reproduce it with the alpha, thanks, but not the 22.12 release on FFmpeg 5.1.2 or my local machine with FFmpeg 6 (which the Shotcut nightlies already moved onto). It might be related to one of the recent changes and older FFmpeg. I will merge your patch.

**Comment by ddennedy** [comment]
fixed in 417bf4769da19178ddfae48108daaf82cbeb25e6
